centreon/src/Core/HostCategory/Infrastructure/Repository/DbWriteHostCategoryActionLogRepository.php

```diff
@@ -0,0 +1,244 @@
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\HostCategory\Infrastructure\Repository;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\Repository\RepositoryException;
+use Centreon\Infrastructure\DatabaseConnection;
+use Core\ActionLog\Application\Repository\WriteActionLogRepositoryInterface;
+use Core\ActionLog\Domain\Model\ActionLog;
+use Core\Common\Infrastructure\Repository\AbstractRepositoryRDB;
+use Core\HostCategory\Application\Repository\ReadHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\Repository\WriteHostCategoryRepositoryInterface;
+use Core\HostCategory\Domain\Model\HostCategory;
+use Core\HostCategory\Domain\Model\NewHostCategory;
+
+class DbWriteHostCategoryActionLogRepository extends AbstractRepositoryRDB implements WriteHostCategoryRepositoryInterface
+{
+    use LoggerTrait;
+    public const HOST_CATEGORY_PROPERTIES_MAP = [
+        'name' => 'hc_name',
+        'alias' => 'hc_alias',
+        'comment' => 'hc_comment',
+        'isActivated' => 'hc_activate',
+    ];
+
+    public function __construct(
+        private readonly WriteHostCategoryRepositoryInterface $writeHostCategoryRepository,
+        private readonly WriteActionLogRepositoryInterface $writeActionLogRepository,
+        private readonly ReadHostCategoryRepositoryInterface $readHostCategoryRepository,
+        private readonly ContactInterface $user,
+        DatabaseConnection $db
+    ) {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function deleteById(int $hostCategoryId): void
+    {
+        try {
+            $hostCategory = $this->readHostCategoryRepository->findById($hostCategoryId);
+            if ($hostCategory === null) {
+                throw new RepositoryException('Cannot find host category to delete.');
+            }
+            $this->writeHostCategoryRepository->deleteById($hostCategoryId);
+            $actionLog = new ActionLog(
+                objectType: 'hostcategory',
+                objectId: $hostCategoryId,
+                objectName: $hostCategory->getName(),
+                actionType: ActionLog::ACTION_TYPE_DELETE,
+                contactId: $this->user->getId(),
+            );
+            $this->writeActionLogRepository->addAction($actionLog);
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+
+            throw $ex;
+        }
```

Comment:
AGR has suggested in my PR to add logs in every catch, I liked the idea for an better debug experience 😃 
```suggestion
        } catch (\Throwable $ex) {
            CentreonLog::create()->error(
                logTypeId: CentreonLog::TYPE_BUSINESS_LOG,
                message: "Error while deleting host category by api : {$th->getMessage()}",
                customContext: ['host_category_id' => $hostCategoryId, 'base_path' => $basePath],
                exception: $th
            );
            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);

            throw $ex;
        }
```